### PR TITLE
Make casings not drop on your feet

### PR DESCRIPTION
**Pull request by Arendian**
## About the PR

Makes bullet casings launch a bit away from you when shooting a gun. Cycling still drops it at your feet.

**Media**

https://github.com/space-wizards/space-station-14/assets/137322659/daf026b1-c8eb-49e5-be9c-185b99ddf424

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: Dygon
- tweak: When firing a gun, casings now eject with a bit of force instead of dropping straight to the floor.

### CODE REVIEW

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs [hunk 1]
```diff
@@ -389,6 +390,12 @@ protected void SetCartridgeSpent(EntityUid uid, CartridgeAmmoComponent cartridge
         xform.LocalRotation = Random.NextAngle();
         xform.Coordinates = coordinates;
 
+        // decides direction the casing ejects and only when not cycling
+        if (angle != 0f)
```
Comment (EmoGarbage404) [inline Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:394]: just make the angle nullable, in that case.

**Comment by Flareguy** [comment]
I feel like these should eject at a much higher speed, right now it looks kind of strange - almost like they're being ejected in slow-motion.

Also, this is player-facing, no? It should probably have a CL.

**Comment by Arendian** [comment]
> I feel like these should eject at a much higher speed, right now it looks kind of strange - almost like they're being ejected in slow-motion.

Do you think the new speed is fast enough?

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs (hunk 1 above)
Comment (Arendian) [reply Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:394]: When I try to make it nullable I get this:
image

**Comment by hord-brayden** [comment]
I like this for realism, but also for leaving evidence. IRL revolvers are sometimes preferable for crime because they don't throw little pieces of evidence (casings) everywhere. You can choose where to dump them all out, usually at a safer location for proper disposal.

**Comment by Flareguy** [comment]
> > I feel like these should eject at a much higher speed, right now it looks kind of strange - almost like they're being ejected in slow-motion.
> 
> I updated the video in the PR, do you think the new ejection speed is fast enough?

Looks fine for the most part. I doubt you could get it to look much better, at least.

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:394]: Use null as default.

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs [hunk 2]
```diff
@@ -389,6 +390,12 @@ protected void SetCartridgeSpent(EntityUid uid, CartridgeAmmoComponent cartridge
...
+        {
+            angle += 3.7f;
+            ThrowingSystem.TryThrow(entity, angle.ToVec().Normalized() / 100, 5f);
```
Comment (metalgearsloth) [inline Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:397]: wut

Comment (Arendian) [reply Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:397]: the angle value adds 212 degrees to the direction someone aims at, making the bullet casing fly in a bit of an angle from the opposite of where they are shooting. 
The /100 is a scuffed way of making the bullets drop to the floor a little bit faster and can be removed if you don't like the look of it. 
The 5f is the force at which the casings get thrown.

Bullet spread if shooting at one direction with the /100:
image

Bullet spread if shooting at one direction without the /100:
image

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs (hunk 1 above)
Comment (Arendian) [reply Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs:394]: It throws the same errors if I make it null

image
image

Comment (Partmedia) [reply Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs]: I've fixed the null-ness problems.

File: Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs (hunk 2 above)
Comment (Partmedia) [reply Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs]: I added a comment to the code that explains this.

**Pull request merged**